Add 'checkout' command

404Wolf opened the pull request:
Add "vt checkout <branch>" command

404Wolf commented on src/cmd/git.ts:
```diff
@@ -67,13 +72,23 @@ const cloneCmd = new Command()
+          "Cannot pull with uncommitted changes. " +
+            "Use `pull -f` to discard local changes.",
+        );
+        return;
```
Do we want to delete new stuff when they force pull or only overwrite files that were modified? Right now it only downloads the current state and doesn't get rid of dirty new files.

404Wolf commented on src/sdk.ts:
```diff
@@ -25,7 +14,7 @@ async function defaultBranchId(projectId: string): Promise<string> {
  * @returns {Promise} Promise resolving to the branch ID
  * @throws {Error} if the branch is not found or if the API request fails
  */
-async function branchIdToName(
```
This was a typo

404Wolf commented on src/vt/git/pull.ts:
```diff
@@ -36,12 +33,6 @@ export async function pull({
     ignoreGlobs,
   });
 
-  if (isDirty(statusResult)) {
```
This is not the job of /git/pull.ts to deal with checking dirtyness -- that should be done by the CLI (and the user can say they're cool with it anyway)

404Wolf commented on src/vt/git/tests/cases.ts:
```diff
@@ -3,58 +3,81 @@ import { ExpectedProjectInode } from "~/vt/git/tests/utils.ts";
 export const testCases: TestCase[] = [
   {
     name: "/Wolf's vtTestProj",
     projectId: "b95fe488-f4de-11ef-97f1-569c3dd06744",
```
Had to refactor so that a test case can contain multiple branches :/

404Wolf commented on src/vt/git/tests/checkout_test.ts:
```diff
@@ -0,0 +1,126 @@
+      }
+    } finally {
+      await cleanup();
+      // TODO `await sdk.projects.delete(project.id);` (this API endpoint
```
Pollutes the test account for now

maxmcd commented:
It looks like this will tell me there are local changes even if I `touch $file` but the content is the same? Or maybe another issue, but if I clone, checkout, touch a file, and then checkout another branch it gives me `X Cannot checkout with uncommitted changes. Use `checkout -f` to ignore local changes.`,

maxmcd commented:
@404Wolf should this not treat added and uncommitted files as dirty? I would have assumed that a file sitting in the working directly is ok and I could checkout around it. I am a bit worried that I am just expecting git functionality, wdyt?

404Wolf commented:
@maxmcd 

It looks like there was a bug that had to do with how I was retrieving files -- the intended behavior (which I think I've fixed now) is that you've only "modified" a file if the content is different, and the mtime is only used as a shortcut to see whether we need to actually cross check the content. I added a test for this too.

I agree, only modified files should "dirty" the state (so deletions shouldn't either. I've updated this :). Later I can add better reporting to this.

![fce94002-d039-4ae8-b5f3-cc64dfd84abd](https://github.com/user-attachments/assets/5e8770ad-8862-41d8-8db2-608b3d6b6d37)
